src/main/java/com/jcraft/jsch/Slf4jLogger.java

```diff
@@ -1,48 +1,63 @@
 package com.jcraft.jsch;
 
+import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class Slf4jLogger implements com.jcraft.jsch.Logger {
 
-  private static final org.slf4j.Logger logger = LoggerFactory.getLogger(JSch.class);
+  private static final Logger stlogger = LoggerFactory.getLogger(JSch.class);
+  private final Logger logger;
 
-  public Slf4jLogger() {}
+  public Slf4jLogger() {
+    this(stlogger);
+  }
+
+  Slf4jLogger(Logger logger) {
+    this.logger = logger;
+  }
 
   @Override
   public boolean isEnabled(int level) {
-    switch (level) {
-      case com.jcraft.jsch.Logger.DEBUG:
-        return logger.isDebugEnabled();
-      case com.jcraft.jsch.Logger.INFO:
-        return logger.isInfoEnabled();
-      case com.jcraft.jsch.Logger.WARN:
-        return logger.isWarnEnabled();
-      case com.jcraft.jsch.Logger.ERROR:
-      case com.jcraft.jsch.Logger.FATAL:
-        return logger.isErrorEnabled();
-      default:
-        return logger.isTraceEnabled();
+    switch(level) {
+        case com.jcraft.jsch.Logger.DEBUG:
+            return logger.isDebugEnabled();
+        case com.jcraft.jsch.Logger.ERROR:
+        case com.jcraft.jsch.Logger.FATAL:
+            return logger.isErrorEnabled();
+        case com.jcraft.jsch.Logger.INFO:
+            return logger.isInfoEnabled();
+        case com.jcraft.jsch.Logger.WARN:
+            return logger.isWarnEnabled();
     }
+    return logger.isTraceEnabled();
   }
 
   @Override
   public void log(int level, String message) {
+    log (level, message, null);
+  }
+
+  @Override
+  public void log(int level, String message, Throwable cause) {
+    if (!isEnabled(level)) {
+      return;
+    }
     switch (level) {
       case com.jcraft.jsch.Logger.DEBUG:
-        logger.debug(message);
+        logger.debug(message, cause);
+        break;
+      case com.jcraft.jsch.Logger.ERROR:
+      case com.jcraft.jsch.Logger.FATAL:
+        logger.error(message, cause);
         break;
       case com.jcraft.jsch.Logger.INFO:
-        logger.info(message);
+        logger.info(message, cause);
         break;
       case com.jcraft.jsch.Logger.WARN:
-        logger.warn(message);
-        break;
-      case com.jcraft.jsch.Logger.ERROR:
-      case com.jcraft.jsch.Logger.FATAL:
-        logger.error(message);
+        logger.warn(message, cause);
         break;
       default:
-        logger.trace(message);
+        logger.trace(message, cause);
```

Comment:
Can we order the cases in this switch statement to match original, which was by severity?